### PR TITLE
feat(buffer_worker): set default queue mode to `memory_only`

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://emqx.atlassian.net/browse/EMQX-9367

For better user experience and performance for the average bridge, we
should change the default queue mode to `memory_only`, as was the
behavior of most bridges in e4.x.  This leads to better performance
when message rate is high enough and the remote resource is not
keeping up with EMQX.

Also, we set the default segment size to equal max queue bytes, 
and the default max queue bytes to 256 MB.

### <samp>🤖 Generated by Copilot at 1e698f3</samp>

This pull request enhances the resource buffer worker module with more flexible and efficient queue options. It adds a `queue_mode` option to choose between memory-only or volatile-offload queue modes, and a `queue_seg_bytes` option to control the disk segment size for the latter mode. It also updates the resource schema module, the test suite, and the translation file to support and validate these new options. It increases the default queue size for the resource buffer worker to 256MB.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [X] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [X] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [X] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [na] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [na] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: rel/i18n/emqx_resource_schema.hocon [hunk 1]
```diff
@@ -179,5 +179,43 @@ When disabled the messages are buffered in RAM only."""
...
+          "<code>memory_only</code>: 所有的消息都缓存在内存里。"
+          " 如果 EMQX 服务重启，缓存的消息会丢失。\n"
+          "<code>hybrid</code>: 先将消息缓存在内存中，当内存中的消息堆积超过一定限制"
+          "（配置项 <code>queue_seg_bytes</code> 描述了该限制）后，"
+          " 后续的消息会缓存到磁盘上。与 <code>memory_only</code> 模式一样，"
+          " 如果 EMQX 服务重启，缓存的消息会丢失。"
+    }
+    label {
+      en: "Queue Mode"
+      zh: "排队模式"
```
Comment (zmstone) [inline rel/i18n/emqx_resource_schema.hocon:216]:
```suggestion
      zh: "缓存模式"
```

File: apps/emqx_resource/src/schema/emqx_resource_schema.erl [hunk 2]
```diff
@@ -149,4 +151,17 @@ max_queue_bytes(default) -> ?DEFAULT_QUEUE_SIZE_RAW;
 max_queue_bytes(required) -> false;
 max_queue_bytes(_) -> undefined.
 
+queue_mode(type) -> enum([memory_only, volatile_offload]);
```
Comment (zmstone) [inline apps/emqx_resource/src/schema/emqx_resource_schema.erl:154]: rename 'queue' to buffer everywhere?
the max_queue_bytes should be named max_buffer_bytes, and move 'max_queue_bytes' to an aliase.

File: rel/i18n/emqx_resource_schema.hocon [hunk 3]
```diff
@@ -179,5 +179,43 @@ When disabled the messages are buffered in RAM only."""
...
+          "\n"
+          "<code>memory_only</code>: 所有的消息都缓存在内存里。"
+          " 如果 EMQX 服务重启，缓存的消息会丢失。\n"
+          "<code>hybrid</code>: 先将消息缓存在内存中，当内存中的消息堆积超过一定限制"
+          "（配置项 <code>queue_seg_bytes</code> 描述了该限制）后，"
+          " 后续的消息会缓存到磁盘上。与 <code>memory_only</code> 模式一样，"
+          " 如果 EMQX 服务重启，缓存的消息会丢失。"
+    }
+    label {
+      en: "Queue Mode"
```
Comment (zmstone) [inline rel/i18n/emqx_resource_schema.hocon:215]:
```suggestion
      en: "Buffer Mode"
```

File: rel/i18n/emqx_resource_schema.hocon [hunk 4]
```diff
@@ -179,5 +179,43 @@ When disabled the messages are buffered in RAM only."""
...
+          " EMQX node restart."
+      zh: "队列操作模式。\n"
+          "\n"
+          "<code>memory_only</code>: 所有的消息都缓存在内存里。"
+          " 如果 EMQX 服务重启，缓存的消息会丢失。\n"
+          "<code>hybrid</code>: 先将消息缓存在内存中，当内存中的消息堆积超过一定限制"
+          "（配置项 <code>queue_seg_bytes</code> 描述了该限制）后，"
+          " 后续的消息会缓存到磁盘上。与 <code>memory_only</code> 模式一样，"
+          " 如果 EMQX 服务重启，缓存的消息会丢失。"
+    }
```
Comment (zmstone) [inline rel/i18n/emqx_resource_schema.hocon:213]: 1. hybrid -> volatile_offload.
2. there is no need to discuss the buffer durability here in this config.

```suggestion
      en: "Queue operation mode.\n"
          "<code>memory_only</mode>: Buffer all messages in memory."
          "<code>volatile_offload</code>: Buffer message in memory first, when up to certain limit"
          " (see <code>buffer_seg_bytes</code> config for more information), then start offloading messages to disk"
      zh: "队列操作模式。\n"
          "<code>memory_only</code>: 所有的消息都缓存在内存里。"
          "<code>volatile_offload</code>: 先将消息缓存在内存中，当内存中的消息堆积超过一定限制"
          "（配置项 <code>buffer_seg_bytes</code> 该限制）后，"
          " 消息会缓存到磁盘上"
    }
```

File: apps/emqx_resource/src/schema/emqx_resource_schema.erl (hunk 2 above)
Comment (thalesmg) [reply apps/emqx_resource/src/schema/emqx_resource_schema.erl]: :heavy_check_mark:

File: rel/i18n/emqx_resource_schema.hocon [hunk 5]
```diff
@@ -179,5 +179,36 @@ When disabled the messages are buffered in RAM only."""
...
+  buffer_mode {
+    desc {
+      en: "Buffer operation mode.\n"
+          "<code>memory_only</mode>: Buffer all messages in memory."
+          "<code>volatile_offload</code>: Buffer message in memory first, when up to certain limit"
+          " (see <code>buffer_seg_bytes</code> config for more information), then start offloading messages to disk"
+      zh: "队列操作模式。\n"
+          "<code>memory_only</code>: 所有的消息都缓存在内存里。"
+          "<code>volatile_offload</code>: 先将消息缓存在内存中，当内存中的消息堆积超过一定限制"
+          "（配置项 <code>buffer_seg_bytes</code> 该限制）后，"
```
Comment (zmstone) [inline rel/i18n/emqx_resource_schema.hocon:204]:
```suggestion
          "（配置项 <code>buffer_seg_bytes</code> 指定该限制）后，"
```

File: rel/i18n/emqx_resource_schema.hocon [hunk 6]
```diff
@@ -179,5 +179,36 @@ When disabled the messages are buffered in RAM only."""
...
+          " 消息会缓存到磁盘上"
```
Comment (zmstone) [inline rel/i18n/emqx_resource_schema.hocon:205]:
```suggestion
          " 消息会开始缓存到磁盘上。"
```

**Comment by zmstone** [comment]
last two comments. LGTM otherwise.

**Review by zmstone — APPROVED** [review]

**Pull request merged**